class.photon.php

```diff
@@ -473,6 +473,34 @@ public function filter_image_downsize( $image, $attachment_id, $size ) {
 			return $image;
 		}
 
+		// Similar to the need to skip the admin, we should skip the REST API to avoid Photonizing images out of context.
+		if ( defined( 'REST_REQUEST' ) && REST_REQUEST ) {
+			global $wp;
+			if ( false !== strpos( $wp->query_vars['rest_route'], '/wp/v2/media/' ) && ! empty( $_REQUEST['context'] ) && 'edit' === $_REQUEST['context'] && // phpcs:ignore WordPress.Security.NonceVerification.NoNonceVerification
```

Comment:
Rather than all these conditionals, it may be simpler to attach something to the `rest_request_before_callbacks` filter (and cleanup via the `rest_request_after_callbacks` filter).

Reply:
The `WP_REST_Request` object is passed to these filters, so we could more robustly check the `context` (which isn't always set via `$_GET` parameter: sometimes it's set programmatically, e.g., when calling `rest_do_request()`).

Reply:
Thank you! I thought there had to be a more elegant way of handling it and the callback filter was the missing link.

Reply:
Changed things up to use the `rest_request_before_callbacks`. I'm chewing on still how to clean it up by the way I did it.